website/addons/twofactor/views.py

```diff
@@ -25,3 +26,37 @@ def user_settings(user_addon, *args, **kwargs):
         message_short='Forbidden',
         message_long='The two-factor verification code you provided is invalid.'
     ))
+
+@must_be_logged_in
+def twofactor_settings_get(auth, *args, **kwargs):
+    return {
+        'result': serialize_settings(auth),
+    }
+
+
+@must_be_logged_in
+def twofactor_enable(auth, *args, **kwargs):
+    user = auth.user
+
+    if user.has_addon('twofactor'):
+        return HTTPError(http.BAD_REQUEST, message='This user already has twofactor enabled')
+
+    user.add_addon('twofactor', auth=auth)
+    user_addon = user.get_addon('twofactor')
+    user_addon.save()
+    user.save()
+    return {
+        'result': serialize_settings(auth),
+    }
+
+@must_be_logged_in
+@must_have_addon('twofactor', 'user')
+def twofactor_disable(auth, *args, **kwargs):
+
+    if auth.user.delete_addon('twofactor', auth=auth):
+        auth.user.save()
+        return {}
+    else:
+        raise HTTPError(http.INTERNAL_SERVER_ERROR, data=dict(
+            message='Could not disable twofactor at this time'
```

Comment:
The key should be `message_long` here. Also check casing and hyphenation of "twofactor"